src/build_log.cc

```diff
@@ -279,9 +279,11 @@ LoadStatus BuildLog::Load(const string& path, string* err) {
     if (!log_version) {
       sscanf(line_start, kFileSignature, &log_version);
 
-      if (log_version < kOldestSupportedVersion) {
-        *err = ("build log version invalid, perhaps due to being too old; "
-                "starting over");
+      if (log_version < kOldestSupportedVersion ||
+          log_version > kCurrentVersion) {
+        *err =
+            ("build log version invalid, perhaps due to being too old or "
+             "too new; starting over");
```

Comment:
Can you turn this into two error messages? One for too old and one for too new.

Reply:
I updated it. PTAL.